packages/react-checkbox/src/components/Checkbox/useCheckbox.tsx

```diff
@@ -28,45 +26,57 @@ const mergeProps = makeMergeProps<CheckboxState>({ deepMerge: checkboxShorthandP
  * @param props - props from this instance of Checkbox
  * @param ref - reference to root HTMLElement of Checkbox
  */
-export const useCheckbox = (props: CheckboxProps, ref: React.Ref<HTMLElement>): CheckboxState => {
-  const state: CheckboxState = mergeProps(
-    {
-      ref,
-      as: props.children !== undefined ? Label : 'span',
-      id: useId('checkbox-'),
-      size: 'medium',
-      labelPosition: 'after',
-      input: {
-        as: 'input',
-        type: 'checkbox',
-        children: null,
-      },
-      indicator: {
-        as: 'div',
-      },
-    },
-    resolveShorthandProps(props, checkboxShorthandProps),
-  );
-
+export const useCheckbox = (
+  {
+    disabled = false,
+    circular = false,
+    required = false,
+    id,
+    defaultChecked,
+    checked: propsChecked,
+    ...props
+  }: CheckboxProps,
```

Comment:
- Please do not use `...rest` as props as then it could be easy to loose some required DOM attributes
- nit: Please move this destruction into body of the function

⬇⬇⬇

```tsx
export const useCheckbox = (
  props: CheckboxProps,
  ref: React.Ref<HTMLElement>
): CheckboxState => {
  const {
    disabled = false,
    circular = false,
    required = false,
    id,
    defaultChecked,
    checked: propsChecked,
    ...rest
  } = props;
  // ...
};
```